src/extra/components/adq.py

```diff
@@ -0,0 +1,1294 @@
+        return self._clock_ratio
+
+    @property
+    def first_pulse_offset(self):
+        """Sample position of first pulse on digitizer trace."""
+        return self._first_pulse_offset
+
+    @property
+    def single_pulse_length(self):
+        """Length in samples in case of a single pulse."""
+        return self._single_pulse_length
+
+    @property
+    def sampling_rate(self):
+        """Sampling rate in Hz."""
+        return AdqChannel._bunch_repetition_rate * self._clock_ratio
+
+    @property
+    def sampling_period(self):
+        """Period between samples in seconds."""
+        return 1 / self.sampling_rate
+
+    @property
+    def trace_shape(self):
+        """Shape of a single trace."""
+        return self._raw_key.entry_shape[0]
+
+    @property
+    def trace_duration(self):
+        """Duration of a single trace in seconds."""
+        return self.trace_shape * self.sampling_period
+
+    @property
+    def board_parameters(self):
+        """All board-related parameters."""
+
+        if self._control_src is None:
+            raise ValueError('component must be initialized with control '
+                             'source data')
+
+        board_node = f'board{self.board}'
+        board_vals = {}
+
+        for k, v in self._control_src.run_values().items():
+            if k.endswith('.timestamp') or not k.startswith(board_node):
+                # Skip timestamps and non-board nodes.
+                continue
+
+            if '.diag.' in k or '.temperature.' in k or 'channel_' in k:
+                # Skip diagnostics, temperatures or channel-specifics.
+                continue
+
+            # Strip board prefix and .value suffix.
+            board_vals[k[7:-6]] = v
+
+        return board_vals
+
+    @property
+    def channel_parameters(self):
+        """All channel-related parameters."""
+
+        if self._control_src is None:
+            raise ValueError('component must be initialized with control '
+                             'source data')
+
+        board_node = f'board{self.board}'
+        channel_node = f'channel_{self.number}'
+        channel_vals = {}
+
+        for k, v in self._control_src.run_values().items():
+            if k.endswith('.timestamp') or not k.startswith(board_node):
+                # Skip timestamps and non-board nodes.
+                continue
+
+            if '.diag.' in k or '.temperature.' in k:
+                # Skip diagnostics and temperatures.
+                continue
+
+            if channel_node in k:
+                # Strip channel node if present
+                k = k.replace(f'.{channel_node}', '')
+            elif 'channel_' in k:
+                # Skip other channel's settings.
+                continue
+
+            # Strip board prefix and .value suffix.
+            channel_vals[k[7:-6]] = v
+
+        return channel_vals
+
+    def select_trains(self, trains):
```

Comment:
Actually our components generally don't have that so far 🤔 

But I agree it would be useful, albeit sounds like something we could provide generically for anything that has `.select_trains()` already, potentially to be overwritten if more knowledge exists.